src/app/shared/provider.service.ts

```diff
@@ -24,9 +27,31 @@ export class ProviderService {
 
     tool.provider = this.getProvider(gitUrl);
     tool.providerUrl = this.getProviderUrl(gitUrl, tool.provider);
+    tool.providerIcon = this.getProviderIcon(tool.gitUrl);
     return tool;
   }
 
+  private getProviderIcon(providerUrl: string): IconDefinition {
+    const provider = this.getProvider(providerUrl);
+    if (!provider) {
+      return null;
+    }
+    if (provider === 'GitHub') {
```

Comment:
Should have mentioned this in my previous comment, but ideally you would create constants out of these strings so they are not duplicated here and in the getProvider method.